workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContent.tsx

```diff
@@ -120,21 +112,77 @@ export const WorkflowRunsTabContent = () => {
                 capitalize(data.category ?? VALUE_UNAVAILABLE),
             },
           ]),
-      { title: 'Started', field: 'started', defaultSort: 'desc' },
+      { title: 'Started', field: 'start', defaultSort: 'desc' },
       { title: 'Duration', field: 'duration' },
     ],
     [workflowInstanceLink, workflowId],
   );
 
-  const filteredData = React.useMemo(
-    () =>
-      (value ?? []).filter(
-        (row: WorkflowRunDetail) =>
-          statusSelectorValue === Selector.AllItems ||
-          row.status === statusSelectorValue,
-      ),
-    [statusSelectorValue, value],
+  const [page, setPage] = useState(0);
+  const [pageSize, setPageSize] = useState(DEFAULT_TABLE_PAGE_SIZE);
+  const [orderBy, setOrderBy] = useState<number>(
+    columns.findIndex(value => value.field === 'start'),
   );
+  const [orderDirection, setOrderDirection] = useState('desc');
+
+  const getFilter = React.useCallback((): Filter | undefined => {
+    const statusFilter: FieldFilter | undefined =
+      statusSelectorValue !== Selector.AllItems
+        ? {
+            operator: 'EQ',
+            value: statusSelectorValue,
+            field: 'state',
+          }
+        : undefined;
+    const workflowIdFilter: FieldFilter | undefined = workflowId
+      ? {
+          operator: 'EQ',
+          value: workflowId,
+          field: 'processId',
+        }
+      : undefined;
+    if (statusFilter && workflowIdFilter) {
+      return {
+        operator: 'AND',
+        filters: [statusFilter, workflowIdFilter],
+      };
+    } else if (statusFilter) {
+      return statusFilter;
+    } else if (workflowIdFilter) {
+      return workflowIdFilter;
+    }
+    return undefined;
+  }, [statusSelectorValue, workflowId]);
+
+  const fetchInstances = React.useCallback(async () => {
+    const paginationInfo: PaginationInfoDTO = {
+      pageSize: pageSize + 1, // add one more to know if to disable the next page
```

Comment:
can you explain a bit this?

Reply:
I need to know if there's more instances then what's shown now, in order to know if to disable the next button or not
Otherwise the user will click on next and there will be no more instances

Reply:
I improved the comment, is it more clear now?